src/client/common/client_common.cpp

```diff
@@ -191,51 +190,9 @@ void mp::client::register_global_settings_handlers()
 }
 
 std::shared_ptr<grpc::Channel> mp::client::make_channel(const std::string& server_address,
-                                                        mp::CertProvider* cert_provider)
+                                                        const mp::CertProvider& cert_provider)
```

Comment:
Yeah, we could've kept the pointer but since after this refactoring we never pass a null pointer to this function, we can use a reference instead for more safety